state/in_memory_store.go

```diff
@@ -20,7 +20,7 @@ type InMemoryStore struct {
 // configStorage is the storage for the configuration with its own mutex lock
 type configStorage struct {
 	mu   sync.RWMutex
-	conf config.Config
+	conf *config.Config
```

Comment:
I'm a bit confused on your point 

`Store config.Config as a pointer - this allows modifications to the config
 down the line`

Can you give me a scenario? If you don't store the config as a pointer, but the method altering it takes a pointer receiver (i.e. `s *InMemoryStory), won't it still alter the underlying config?

Reply:
Yes, you're right. Thank you for catching this